### DISCUSSION TITLE
Welcome to D3SOX Vencord userplugins Discussions!

### DISCUSSION THREAD

**Discussion by D3SOX** [D3SOX/vencord-userplugins#22] [Announcements]
## 👋 Welcome!
You can use discussions to
  * Ask questions about my plugins you’re wondering about.
  * Share improvements for the plugins or suggest similar ones.

**Comment by Chaython** [comment]
I enjoy 'voiceactivity' on betterdiscord unlike user voice show... it show's an Icon next to every user that is in a vc anywhere....
https://betterdiscord.app/plugin/voiceactivity it's really helpful for finding who's actually on...
Currently as I've switched to vencord, I've just been going down my list right click following everyone....
Would it be possible to integrate the two together? So It shows the icon and single click to follow them?

**Reply by Chaython** [reply]
I also think its funny to fake deafen, I really liked this version, but it no longer worked on newer versions of bd because the library was out of date https://github.com/YofukashiNo/BetterDiscordPlugins/blob/master/FakeDeafen.plugin.js
So I currently use
```
var text = new TextDecoder("utf-8");
var _fakeDeafen = false;
var _keyUpEvent = true;

WebSocket.prototype.original = WebSocket.prototype.send;
WebSocket.prototype.send = function(data) {
	if (_fakeDeafen) {
		if (Object.prototype.toString.call(data) === "[object ArrayBuffer]") {
			if (text.decode(data).includes("self_deaf")) {
				console.log("found mute/deafen data");
				data = data.replace('"self_mute":false', 'NiceOneDiscord');
				console.log("faked - borkgang.com");
			}
		}
	}
	WebSocket.prototype.original.apply(this, [data]);
}

// ------- //

var _keyUpEventFunc = function(e) {
	var keyLocation = ["Standard", "Left", "Right", "Numpad", "Mobile", "Joystick"][e.location];
	if (e.key == "Alt" && keyLocation == "Right") {
		if(document.querySelectorAll('[aria-label="Deafen"')[0].ariaChecked == "false") {
			document.querySelectorAll('[aria-label="Deafen"')[0].click()
			setTimeout(() => { _fakeDeafen ? _fakeDeafen = false : _fakeDeafen = true;; }, 100);
			setTimeout(() => { document.querySelectorAll('[aria-label="Deafen"')[0].click(); }, 100);
			console.log("Toggled fake deafen: " + _fakeDeafen)
			return;
		}
		
		_fakeDeafen ? _fakeDeafen = false : _fakeDeafen = true;
		document.querySelectorAll('[aria-label="Deafen"')[0].click()
		console.log("Toggled fake deafen: " + _fakeDeafen)
	}
}

document.addEventListener("keyup", function(e) {
	if (_keyUpEvent) {
		_keyUpEventFunc(e);
	}
}, false);
```

**Reply by D3SOX** [reply]
First and foremost: This is not the official Vencord plugin request repo so you might suggest things but I only implement them when I really like the idea. It's best to always open these at https://github.com/Vencord/plugin-requests/issues because there are more eyes on that.

As for VoiceActivity there are already 2 PRs which I've been using for quite some time:
https://github.com/Vendicated/Vencord/pull/2248
https://github.com/Vendicated/Vencord/pull/1669

I don't support fake voice actions but if you look around GitHub you can definitely find a plugin for that

As a side note you shouldn't use announcements but rather create a separate topic in the appropriate category
